packages/apollo-federation/src/composition/validate/sdl/__tests__/uniqueTypeNamesWithoutEnumsOrScalars.test.ts

```diff
@@ -0,0 +1,487 @@
+import {
+  GraphQLSchema,
+  specifiedDirectives,
+  Kind,
+  DocumentNode,
+} from 'graphql';
+import { validateSDL } from 'graphql/validation/validate';
+import gql from 'graphql-tag';
+import {
+  typeSerializer,
+  graphqlErrorSerializer,
+} from '../../../../snapshotSerializers';
+import federationDirectives from '../../../../directives';
+import { UniqueTypeNamesWithoutEnumsOrScalars } from '..';
+import { ServiceDefinition } from '../../../types';
+import { buildMapsFromServiceList } from '../../../compose';
+
+expect.addSnapshotSerializer(graphqlErrorSerializer);
+expect.addSnapshotSerializer(typeSerializer);
+
+function createDocumentsForServices(
+  serviceList: ServiceDefinition[],
+): DocumentNode[] {
+  const { definitionsMap, extensionsMap } = buildMapsFromServiceList(
+    serviceList,
+  );
+  return [
+    {
+      kind: Kind.DOCUMENT,
+      definitions: Object.values(definitionsMap).flat(),
+    },
+    {
+      kind: Kind.DOCUMENT,
+      definitions: Object.values(extensionsMap).flat(),
+    },
+  ];
+}
+
+describe('UniqueTypeNamesWithoutEnumsOrScalars', () => {
+  let schema: GraphQLSchema;
+
+  // create a blank schema for each test
+  beforeEach(() => {
+    schema = new GraphQLSchema({
+      query: undefined,
+      directives: [...specifiedDirectives, ...federationDirectives],
+    });
+  });
+
+  describe('enforces unique type names for', () => {
+    it('object type definitions (non-identical, non-value types)', () => {
```

Comment:
nit: the tests should ideally read as sentences. Like `it('errors with duplicate type definitions for non-identical non-value types')`.

You don't really have to fix this one, just a matter of preference. It makes it a little easier to read and understand what's going on in failed tests

Reply:
I'd argue that this structure does that well when a test fails:
> UniqueTypeNamesWithFields › enforces unique type names for › object type definitions (non-identical, value types with type mismatch)

But! I would say that the structure while navigating the file can be a bit cumbersome (though collapsing by `describe` works nicely).

Reply:
My test failure workflow would be to find all references to the last bit
> object type definitions (non-identical, value types with type mismatch)

Which isn't perfect in the case that I've duplicated that string (and I have)